cli/src/main/kotlin/OrtServerMain.kt

```diff
@@ -65,18 +67,36 @@ class OrtServerMain : SuspendingNoOpCliktCommand(COMMAND_NAME) {
 
         subcommands(AuthCommand(), RunsCommand())
 
+        @Suppress("ForbiddenMethodCall")
         versionOption(
             version = ORT_SERVER_VERSION,
             names = setOf("--version", "-v"),
-            help = "Show the version and exit.",
-            // TODO: Also add the server version given by `baseUrl` to this output.
-            message = { "$commandName version $it" }
+            help = "Show the version of the CLI and the ORT Server if authenticated.",
+            message = { runBlocking { buildVersionInformation(it) } }
         )
     }
 
     override fun help(context: Context) = """
         The ORT Server Client (OSC) is a Command Line Interface (CLI) to interact with an ORT Server instance.
     """.trimIndent()
+
+    /**
+     * Build the version information for the CLI and (if authenticated) for the ORT Server.
+     */
+    private suspend fun buildVersionInformation(cliVersion: String): String {
+        val serverVersions = runCatching { createOrtServerClient()?.versions?.getVersions() }.getOrNull()
+
+        return buildString {
+            appendLine("$commandName version $cliVersion")
+            if (serverVersions != null) {
+                val serverVersion = serverVersions["ORT Server"]
+                val ortVersion = serverVersions["ORT Core"]
+
+                appendLine("Server version ${serverVersion ?: "<unknown>"}")
+                appendLine("ORT Core version ${ortVersion ?: "<unknown>"}")
```

Comment:
Maybe `<requires authentication>` instead?

Reply:
<unknown> is not printed when the authentication is not available.
It is printed when the authentication was available, but the server returned a map that does not contain `ORT Server`.